lib/diff.ml

```diff
@@ -101,13 +114,37 @@ and module_item ~typing_env ~name ~reference ~current =
       module_declaration ~typing_env ~name ~reference ~current
   | _ -> assert false
 
+and module_type_item ~typing_env ~name ~reference ~current =
+  match (reference, current) with
+  | None, None -> None
+  | None, Some (ModType curr_mtd) ->
+      Some (Modtype { mtname = name; mtdiff = Added curr_mtd })
+  | Some (ModType ref_mtd), None ->
+      Some (Modtype { mtname = name; mtdiff = Removed ref_mtd })
+  | Some (ModType ref_mtd), Some (ModType curr_mtd) ->
+      module_type_declaration ~typing_env ~name ~reference:ref_mtd
+        ~current:curr_mtd
+  | _ -> assert false
+
 and module_declaration ~typing_env ~name ~reference ~current =
-  match (reference.md_type, current.md_type) with
+  module_helper ~typing_env ~name ~ref_module_type:reference.md_type
+    ~current_module_type:current.md_type ~reference_location:reference.md_loc
+
+and module_type_declaration ~typing_env ~name ~reference ~current =
```

Comment:
I think that for the sake of clarity, we should stick to the compiler type names, even though those types have sometimes unclear naming convention.

We have a growing number of functions here so being careful with their names is becoming more important.